packages/manager/cypress/e2e/core/databases/create-database.spec.ts

```diff
@@ -62,13 +62,61 @@ describe('create a database cluster, mocked data', () => {
         });
 
         const clusterSizeSelection =
-          configuration.clusterSize > 1 ? '3 Nodes' : '1 Node';
+          configuration.clusterSize == 1
+            ? '1 Node'
+            : configuration.clusterSize == 2
+            ? '2 Nodes'
+            : '3 Nodes';
+
+        const nodes =
+          configuration.clusterSize == 1
+            ? 'Primary (1 Node)'
+            : configuration.clusterSize == 2
+            ? 'Primary (+1 Node)'
+            : 'Primary (+2 Nodes)';
 
         const clusterCpuType =
           configuration.linodeType.indexOf('-dedicated-') !== -1
             ? 'Dedicated CPU'
             : 'Shared CPU';
 
+        // Function to validate Action Menu on the landing page as per db cluster status
+        const validateActionItems = (state: string) => {
+          const menuStates: Record<string, Record<string, boolean>> = {
+            active: {
+              Delete: true,
+              'Manage Access Controls': true,
+              'Reset Root Password': true,
+              Resize: true,
+              Resume: false,
+              Suspend: true,
+            },
+            provisioning: {
+              Delete: true,
+              'Manage Access Controls': true,
+              'Reset Root Password': true,
+              Resize: true,
+              Resume: false,
+              Suspend: false,
+            },
+          };
+          const expectedItems = menuStates[state];
+          cy.get(`[id="action-menu-for-database-${databaseMock.label}-button"]`)
+            .should('be.visible')
+            .click();
+
+          cy.findByRole('menu')
+            .should('be.visible')
+            .within(() => {
+              Object.entries(expectedItems).forEach(([id, enabled]) => {
+                cy.findByTestId(id)
+                  .should('be.visible')
+                  .should(enabled ? 'be.enabled' : 'be.disabled');
+              });
+            });
```

Comment:
```suggestion
          ui.actionMenu
            .findByTitle(`Action menu for Database ${databaseMock.label}`)
            .should('be.visible')
            .click();

          Object.entries(expectedItems).forEach(([label, enabled]) => {
            ui.actionMenuItem
              .findByTitle(label)
              .should('be.visible')
              .should(enabled ? 'be.enabled' : 'be.disabled');
          });
```
Just a small suggestion: we have the `ui.actionMenu` and `ui.actionMenuItem` helpers to make it easier to select action menu buttons and items.

(If you haven't seen them yet, I definitely encourage you to check out our [UI helpers](https://github.com/linode/manager/tree/develop/packages/manager/cypress/support/ui): they're basically small utils that make it easier to work with some of our common UI elements)

Reply:
Thanks for providing information about UI helpers, I'll definitely use them in my upcoming tests.